File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n=prices.size();
        int buy=INT_MAX;
        int sell=0;
        for(int i=0;i<n;i++){
           // buy=min(buy,prices[i]);
           // sell=max(sell,prices[i]-buy);

           //trying another way to solve
           int num=prices[i];
           if(num<buy){
            buy=num;
           }
           if((num-buy) > sell){
            sell=num-buy;
           }
        }        

        return sell;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy=10**5
        sell=0
        n=len(prices)

        for i in range(0,n):
            num=prices[i]
            if(num<buy):
                buy=num
            
            if((num-buy)>sell):
                sell=num-buy

        
        return sell